config-templates/ibm/ess/2.1.0/deployment.yaml

```diff
@@ -0,0 +1,601 @@
+  # ==================================================================================
+  #  driverRegistrar: "us.gcr.io/k8s-artifacts-prod/sig-storage/csi-node-driver-registrar:v2.0.1"
+
+  # SpectrumScale CSI Plugin image name, in case we do not want to use default image.
+  # ==================================================================================
+  #  spectrumScale: "quay.io/ibm-spectrum-scale/ibm-spectrum-scale-csi-driver:v2.1.0"
+
+  # attacherNodeSelector specifies on which nodes we want to run attacher sidecar
+  # In below example attacher will run on nodes which have label as "scale=true"
+  # and "infranode=2". Can have multiple entries.
+  # ==================================================================================
+    attacherNodeSelector:
+      - key: "scale"
+        value: "true"
+  #    - key: "infranode"
+  #      value: "2"
+
+  # provisionerNodeSelector specifies on which nodes we want to run provisioner
+  # sidecar. In below example provisioner will run on nodes which have label as
+  # "scale=true" and "infranode=1". Can have multiple entries.
+  # ==================================================================================
+    provisionerNodeSelector:
+      - key: "scale"
+        value: "true"
+  #    - key: "infranode"
+  #      value: "1"
+
+  # pluginNodeSelector specifies nodes on which we want to run plugin daemoset
+  # In below example plugin daemonset will run on nodes which have label as
+  # "scale=true". Can have multiple entries.
+  # ==================================================================================
+    pluginNodeSelector:
+      - key: "scale"
+        value: "true"
+
+  # In case K8s nodes name differs from SpectrumScale nodes name, we can provide
+  # node mapping using nodeMapping attribute. Can have multiple entries.
+  # ==================================================================================
+  #  nodeMapping:
+  #    - k8sNode: "< K8s Node Name >"
+  #      spectrumscaleNode: "< SpectrumScale Node Name >"
+  # In case K8s node name start with number then use following node mapping format.
+  #    - k8sNode: "K8sNodePrefix_< K8s Node Name >"
+  #      spectrumscaleNode: "< SpectrumScale Node Name >"
+    nodeMapping:
+      - k8sNode: "K8sNodePrefix_{{k8-n1-ip}}"
+        spectrumscaleNode: "{{sc-n1-host}}"
+      - k8sNode: "K8sNodePrefix_{{k8-n2-ip}}"
+        spectrumscaleNode: "{{sc-n2-host}}"
+      - k8sNode: "K8sNodePrefix_{{k8-n3-ip}}"
+        spectrumscaleNode: "{{sc-n3-host}}"
+      - k8sNode: "K8sNodePrefix_{{k8-n4-ip}}"
+        spectrumscaleNode: "{{sc-n4-host}}"
+      - k8sNode: "K8sNodePrefix_{{k8-n5-ip}}"
+        spectrumscaleNode: "{{sc-n5-host}}"
+      - k8sNode: "K8sNodePrefix_{{k8-n6-ip}}"
+        spectrumscaleNode: "{{sc-n6-host}}"
+      - k8sNode: "K8sNodePrefix_{{k8-n7-ip}}"
+        spectrumscaleNode: "{{sc-n7-host}}"
+      - k8sNode: "K8sNodePrefix_{{k8-n8-ip}}"
+        spectrumscaleNode: "{{sc-n8-host}}"
```

Comment:
I see all of there parameters are optional and they do not have default value . If not values given by the user it will result in to below yaml that may break the changes . Please refere this quide to configure optinal variables https://github.com/IBM/ibm-satellite-storage/blob/prgavali-guide/DEPLOYMENT_README.md#configuration-for-optionalemptyfalse--variable

```
   nodeMapping:
      - k8sNode: "K8sNodePrefix_"
        spectrumscaleNode: ""
      - k8sNode: "K8sNodePrefix_"
        spectrumscaleNode: ""
```